src/components/ipv-callback/ipv-callback-controller.ts

```diff
@@ -1,8 +1,42 @@
 import { Request, Response } from "express";
 import { ExpressRouteFunc } from "../../types";
+import { ReverificationResultInterface } from "./types";
+import { logger } from "../../utils/logger";
+import { reverificationResultService } from "./reverification-result-service";
+import { BadRequestError } from "../../utils/error";
 
-export function ipvCallbackGet(): ExpressRouteFunc {
+export function ipvCallbackGet(
+  service: ReverificationResultInterface = reverificationResultService()
+): ExpressRouteFunc {
   return async function (req: Request, res: Response) {
-    res.status(200).send("Got to ipv callback");
+    const { email } = req.session.user;
+    const { sessionId, clientSessionId, persistentSessionId } = res.locals;
+
+    const code = req.query.code;
+
+    if (code === undefined) {
+      throw new BadRequestError("Request query missing auth code param", 400);
+    } else if (typeof code !== "string") {
+      throw new BadRequestError("Invalid auth code param type", 400);
+    }
+
+    const result = await service.getReverificationResult(
+      sessionId,
+      clientSessionId,
+      persistentSessionId,
+      req,
+      email,
+      code
+    );
+
+    logger.info(
```

Comment:
Do we need to log out when this is successful or just when a User fails to be re-verified?

Reply:
I think for now useful to log in both cases because this is the only feedback in logs - but when we implement more of the flow we can probably just log in the case of non success